core/jobs/batch_jobs/story_node_jobs.py

```diff
@@ -0,0 +1,162 @@
+# coding: utf-8
+#
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs used for populating story node."""
+
+from __future__ import annotations
+
+import logging
+
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import result
+from typing import Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import datastore_services
+    from mypy_imports import story_models
+    from mypy_imports import topic_models
+
+(story_models, topic_models) = models.Registry.import_models([
+    models.Names.STORY, models.Names.TOPIC])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class PopulateStoryNodeJob(base_jobs.JobBase):
+    """Job that migrates story nodes, for serial chapter launch feature"""
+
+    def _update_story_node(
+            self, story_model: story_models.StoryModel
+    ) -> result.Result[
+        story_models.StoryModel,
+        Tuple[str, Exception]
+    ]:
+        """Populate the new fields in each story node of the StoryModel
+        instance.
```

Comment:
It is unclear what "new" here means. Prefer being as precise in the doc string as possible e.g. what are the fields being populated.

Reply:
Done